Gearman client timeout

hidden-primary-net opened the pull request:
With this patch the `on_fail` callback is called implicitely if the timeout elapsed.
All pending tasks of a taskset are iterated and their `fail` method is invoked with a timeout message.

hidden-primary-net committed:
test: Add prove of missing "on_fail" invocation

The "on_fail" callback is not invoked when a task was executed with a
given "timeout" value.

hidden-primary-net committed:
fix: Mark tasks failed on timeout

Iterates all pending tasks of a taskset and calls their "fail" method
with the reason (timeout).

p-alik commented on t/02-client.t:
```diff
@@ -69,4 +69,48 @@ subtest "js socket", sub {
     ok($gc->_get_random_js_sock());
 };
 
+subtest 'Client: "on_fail" handler is triggered on timeout' => sub {
```
subtest fails if no `gearmand` in `$PATH`. Could you add a skip please. [Here an example](https://github.com/p-alik/perl-Gearman/blob/master/t/02-client.t#L61).

hidden-primary-net replied:
Added a `skip_all` as suggested.

hidden-primary-net committed:
fix: Skip new test if gearmand is not available

@p-alik: Thanks for your hint
(https://github.com/p-alik/perl-Gearman/pull/34#pullrequestreview-84783025),
updated the test accordingly.

The test failed if neither `gearmand` was installed nor `GEARMAND_ADDR`
was set. Now the subtest is skipped completely in this case.

p-alik commented:
Thanks, Marco.
